### PR TITLE
Python: check if multiple versions of a package in a closure

### PR DESCRIPTION
**Pull request by FRidh**
While we try to prevent having multiple versions of Python packages it
still sometimes happens we need them. It also occasionally happens that
we end up with multiple versions of the same package in one closure, and
that is something Python cannot handle.

Currently we check for this after building the package. In most cases we
might be able to catch this already during evaluation time, by asserting
we do not have any duplicate name parts of Python derivations.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
To do:
- [ ]: implement in `python.pkgs.buildPythonPackage`
- [ ]: implement in `python.buildEnv`

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -88,6 +123,9 @@ in {
 
   vowpalwabbit = callPackage ../development/python-modules/vowpalwabbit { pythonPackages = self; };
 
+#   unique = assertUniqueNames [ self.django_1_6.name self.django_1_8.name self.numpy.name ];
+  unique = assertUniquePythonPackages [ self.django_1_6 self.django_1_8 self.numpy ];
+
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: @garbas this is what you meant, right? This we could now implement in the builder.

Comment (garbas) [reply pkgs/top-level/python-packages.nix]: @FRidh 👍 looks like a good start

I suppose this should be done automatically by buildPythonPackage

and sorry for the delay .. i'm slowly getting back to online life 😄

**Comment by matthiasbeyer** [comment]
This is stalled and should probably be closed. @FRidh

**Comment by AndersonTorres** [comment]
> This is stalled and should probably be closed. @FRidh

Thy will be done.

**Pull request closed without merging**